src/cipher/address.go

```diff
@@ -153,12 +167,20 @@ func (addr Address) Verify(key PubKey) error {
 // version is first byte in binary format
 // in printed address its key, version, checksum
 func (addr Address) String() string {
-	return string(base58.Hex2Base58(addr.Bytes()))
+	str, err := b58.EncodeToString(addr.Bytes())
+	if err != nil {
```

Comment:
The library returns `err` if `bytes.Buffer.WriteBytes` returns an error. However, this method never returns an error. See: https://golang.org/pkg/bytes/#Buffer.WriteByte
So `EncodeToString` will never return an error.
Furthermore, that library does not check the `WriteBytes` error for one of its calls, which is inconsistent (although it never needs to check the error).

I'm starting to lean towards having our own base58 implementation based upon one of these libraries, but fixing the minor issues.

Still, in a case like this (where our method does not return an error), it should call `panic(err)` (although `err` will always be `nil`).

Reply:
I could try to fix the issues with the existing library but I am also more inclined towards creating a new library at this point